ngraph/frontend/paddlepaddle/src/model.cpp

```diff
@@ -204,16 +238,38 @@ void InputModelPDPD::InputModelPDPDImpl::setPartialShape (Place::Ptr place, cons
     auto var_place = std::dynamic_pointer_cast<TensorPlacePDPD>(place);
     if (var_place) {
         var_place->setPartialShape(p_shape);
+        return;
+    }
+    auto in_port_place = std::dynamic_pointer_cast<InPortPlacePDPD>(place);
+    if (in_port_place) {
+        in_port_place->getSourceTensorPDPD()->setPartialShape(p_shape);
+        return;
     }
-    // TODO: resolve for port places
+    auto out_port_place = std::dynamic_pointer_cast<OutPortPlacePDPD>(place);
+    if (out_port_place) {
+        out_port_place->getTargetTensorPDPD()->setPartialShape(p_shape);
+        return;
+    }
+    PDPD_ASSERT(false, "Cannot set shape for this place.");
```

Comment:
can we introduce new macro like "PDPD_THROW(message)" to avoid compiler warnings?

Reply:
We can, but what kind of warning compiler prints?

Reply:
It prints "warning: control reaches end of non-void function [-Wreturn-type]"

Reply:
These all are void fuctions

Reply:
Oops, yes, I can see this warning in other functions, not these ones